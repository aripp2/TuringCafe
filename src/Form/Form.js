import React, { Component } from 'react';
import './Form.scss';

class Form extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      date: '',
      time: '',
      number: ''
    }
  }

  changeHandler = (e) => {
    this.setState({ [e.target.name]: e.target.value })
  }

  makeReservation = (e) => {
    e.preventDefault()
    const { addReservation } = this.props
    const details = {...this.state, id: Date.now()}
    addReservation(details)

  }

  render() {

    return (
      <form className='resy-form'>
        <input 
          type='text'
          name='name'
          value={this.state.name}
          placeholder='Name'
          onChange={this.changeHandler}
        />
        <input 
          type='text'
          name='date'
          value={this.state.date}
          placeholder='Date (mm/dd)'
          onChange={this.changeHandler}
        />
        <input 
          type='time'
          step='600'
          name='time'
          min="1:00" 
          max="12:00"
          value={this.state.time}
          placeholder='Time'
          onChange={this.changeHandler}
        />
        <input 
          type='number'
          name='number'
          value={this.state.number}
          placeholder='Number of guests'
          onChange={this.changeHandler}
        />
        <button
          onClick={this.makeReservation}
        >Make Reservation</button>
      </form>
      )
  }
}

export default Form;