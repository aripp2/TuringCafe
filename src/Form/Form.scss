

input {
  height: 40px;
  width: 150px;
  margin-right: 20px;
}

button {
  height: 40px;
  width: 150px;
}