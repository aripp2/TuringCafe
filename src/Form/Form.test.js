import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form', () => {
  let wrapper;
  const mockAddRes = jest.fn()
  beforeEach(() => {
    wrapper = shallow(<Form addReservation={mockAddRes}/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})