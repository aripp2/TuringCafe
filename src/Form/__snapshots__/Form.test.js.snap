// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot 1`] = `
<form
  className="resy-form"
>
  <input
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    name="date"
    onChange={[Function]}
    placeholder="Date (mm/dd)"
    type="text"
    value=""
  />
  <input
    max="12:00"
    min="1:00"
    name="time"
    onChange={[Function]}
    placeholder="Time"
    step="600"
    type="time"
    value=""
  />
  <input
    name="number"
    onChange={[Function]}
    placeholder="Number of guests"
    type="number"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Make Reservation
  </button>
</form>
`;
